### ISSUE TITLE
Feature Request/Idea: In Dataverse Guides, indicate which metadatablock properties are required and optional

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse#8854]
**Overview of the Feature Request**

**What kind of user is the feature intended for?**
(Example users roles: API User, Curator, Depositor, Guest, Superuser, Sysadmin)
Curator, Sysadmin

**What inspired the request?**
A TSV file for a new metadatablock being worked on was missing a property that was required. The property in this case was  displayOrder, a property of every TSV file's datasetField section. The metadatablock can't be added to Dataverse installations without the displayOrder property being filled out.

As of v5.11, the guides don't indicate that the displayOrder for each field is required and doesn't indicate which other properties need to be filled and which are optional.

**What existing behavior do you want changed?**
The Dataverse Guides indicate which parts of the TSV file need to be filled out and which are optional. Specifically, which parts of the three sections - metadataBlock, datasetField, and controlledVocabulary - are required or optional.

**Any related open or closed issues to this feature request?**
https://github.com/IQSS/dataverse/issues/7844

**Comment by jggautier** [comment]
The datasetField's description property isn't technically required but I think should be required or recommended since it's what appears when people click or hover over the tooltip icon that's next to the metadata fields. Not including a description means that the tooltip will be empty.

So maybe the guides should indicate which properties are technically required, which are recommended and which are optional.

The [Metadata Customization](https://guides.dataverse.org/en/latest/admin/metadatacustomization.html) page seems like a good place. Maybe adding a line in the "Allowed values and restrictions" column of each property. (Adding a fourth column might make the table less readable, considering how little horizontal space there is now.)

**Comment by pdurbin** [comment]
Thanks for opening this issue, @jggautier 

I recently struggled to get a tsv loaded when it wasn't quite right.

Perhaps this goes without saying but if we're in there making tweaks to the metadata customization page, let's also include any tips that might help someone with the somewhat arduous process of creating and loading a metadata block. tsv.

**Comment by jggautier** [comment]
@pdurbin, do you think the metadata customization page now includes any tips that might help someone create and load a metadata block.tsv? A lot about the page has changed since the v5.11 version we were looking at when I opened this GitHub issue.

Otherwise, I think it's best to open a new GitHub issue describing how we could review the helpfulness of the metadata customization page, such as by talking with folks who've used it recently to create and change metadata blocks, and coming up with and testing changes to the page.

**Comment by pdurbin** [comment]
@jggautier no, there aren't many tips. These days we suggest using Docker.

I'm ok with closing this issue and opening new ones.

**Comment by jggautier** [comment]
Ah, I'm assuming we're suggesting that folks use Docker so that they can more easily create working examples or prototypes that let them experience how they're metadata blocks are working. So maybe that would be a tip, right?

I didn't mean to suggest we close this GitHub issue. I think it'll still be helpful to indicate in the Dataverse Guides which metadata block properties are required and optional.
